File: src/main/java/com/company/idm/domain/rbac/Menu.java
package com.company.idm.domain.rbac;

import com.company.idm.common.enums.MenuType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 菜单领域实体，描述前端可渲染的菜单节点。
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Menu {

    private Long id;
    private String menuCode;
    private String menuName;
    private Long parentId;
    private MenuType menuType;
    private String path;
    private String component;
    private String icon;
    private Integer sortNo;
    private Integer status;
    private Integer visible;
    private Integer minPermissionLevel;
    private String remark;
}

File: frontend/src/types/menu.ts
export interface MenuTreeNode {
  id: number
  menuCode: string
  menuName: string
  menuType: string
  path: string
  component: string
  icon: string
  parentId: number
  sortNo: number
  children: MenuTreeNode[]
}

export interface MenuItem {
  id: number
  menuCode: string
  menuName: string
  parentId: number
  menuType: string
  path: string
  component: string
  icon: string
  sortNo: number
  remark: string | null
}

export interface MenuTreeOption {
  value: number
  label: string
  children: MenuTreeOption[]
}

export interface CreateMenuPayload {
  menuCode: string
  menuName: string
  parentId: number
  menuType: string
  path: string
  component?: string | null
  icon?: string | null
  sortNo: number
  remark?: string | null
}

export interface UpdateMenuPayload {
  menuName: string
  parentId: number
  menuType: string
  path: string
  component?: string | null
  icon?: string | null
  sortNo: number
  remark?: string | null
}
